plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -1,40 +1,50 @@
 package com.google.cloud.tools.eclipse.appengine.localserver.server;
 
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.debug.core.ILaunch;
 import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.jdt.launching.AbstractJavaLaunchConfigurationDelegate;
 import org.eclipse.ui.console.MessageConsole;
-import org.eclipse.ui.console.MessageConsoleStream;
+import org.eclipse.wst.server.core.IModule;
 import org.eclipse.wst.server.core.IServer;
 import org.eclipse.wst.server.core.ServerUtil;
 
 public class LocalAppEngineServerLaunchConfigurationDelegate
-    extends AbstractJavaLaunchConfigurationDelegate {
+extends AbstractJavaLaunchConfigurationDelegate {
 
   @Override
   public void launch(ILaunchConfiguration configuration, String mode, ILaunch launch,
       IProgressMonitor monitor) throws CoreException {
-    
     final IServer server = ServerUtil.getServer(configuration);
     if (server == null) {
       return;
     }
 
+    final IModule[] modules = server.getModules();
+    if (modules == null || modules.length == 0) {
+      return;
+    }
+
     LocalAppEngineServerBehaviour serverBehaviour =
         (LocalAppEngineServerBehaviour) server.loadAdapter(LocalAppEngineServerBehaviour.class, null);
 
-    serverBehaviour.setStarting();
-    
-    // TODO actually launch the server
-    
-    serverBehaviour.setStarted();
-    
-    MessageConsole console = TargetPlatform.findConsole("debugging");
+    List<File> runnables = new ArrayList<File>();
+    for (IModule module : modules) {
+      IPath deployPath = serverBehaviour.getModuleDeployDirectory(module);
+      runnables.add(deployPath.toFile());
+    }
+
+    MessageConsole console = TargetPlatform.findConsole(configuration.getName());
     TargetPlatform.showConsole(console);
-    MessageConsoleStream out = console.newMessageStream();
-    out.println("Server started (not really)");
-  }
+    console.clearConsole();
```

Comment:
are you sure we want to clear the console here? Might there be useful information, possibly completely unrelated to this run, left there that the user will want?